integration-test/Analysis/FixtureExpectationUtils.hs

```diff
@@ -0,0 +1,138 @@
+module Analysis.FixtureExpectationUtils (
+  DependencyResultsSummary (..),
+
+  -- * spec hooks
+  withAnalysisOf,
+
+  -- * expectation helpers
+  getDepResultsOf,
+  shouldFindProjectOf,
+  expectDepResultsSummary,
+  expectNumDeps,
+  expectNumDirects,
+  expectNumEdges,
+  expectDep,
+  expectDirectDep,
+  expectEdgeBetween,
+  expectGraphBreadth,
+  expectNumManifests,
+) where
+
+import Analysis.FixtureUtils (AnalysisTestFixture (..), FixtureArtifact (..), getArtifact, performDiscoveryAndAnalyses)
+import App.Fossa.Analyze.Types (AnalyzeProject)
+import Control.Algebra (Has)
+import Control.Effect.Lift (Lift, sendIO)
+import Control.Monad (unless)
+import Data.List (find, isInfixOf)
+import Data.String.Conversion (toString, toText)
+import Data.Text (Text)
+import Graphing
+import Path
+import Path.IO qualified as PIO
+import Test.Hspec (Expectation, HasCallStack, expectationFailure, shouldNotBe)
+import Types (Dependency (..), DependencyResults (..), DiscoveredProject (..), GraphBreadth)
+
+-- | Tabulated property of the DependencyResult.
+data DependencyResultsSummary = DependencyResultsSummary
+  { numDeps :: Int
+  , numDirectDeps :: Int
+  , numEdges :: Int
+  , numManifestFiles :: Int
+  , graphType :: GraphBreadth
+  }
+  deriving (Show, Eq, Ord)
+
+-- | Performs discovery and analysis for all discovered project fop provided analysis integration fixture.
+withAnalysisOf :: (Has (Lift IO) sig m, AnalyzeProject a, MonadFail m) => AnalysisTestFixture a -> (([(DiscoveredProject a, DependencyResults)], Path Abs Dir) -> m b) -> m ()
+withAnalysisOf a runTest = do
+  extractedDir <- getArtifact (artifact a)
+  res <- performDiscoveryAndAnalyses a
+  _ <- case scopedDir . artifact $ a of
+    Nothing -> runTest (res, extractedDir)
+    Just sd -> runTest (res, extractedDir </> sd)
+  sendIO $ PIO.removeDirRecur extractedDir
+
+-- Expectation Helpers
+-- Provides expectation functions for graph properties and dependency resulting.
+
+-- | Retrieves a dependency result from discovered project of a given type and path.
+getDepResultsOf :: [(DiscoveredProject a, DependencyResults)] -> (Text, Path Abs Dir) -> Maybe DependencyResults
```

Comment:
Some of these I would like to use in future, specifically to ensure we have edge between some dep, and some dep exist as direct (targeting case re deps not being direct or dep missing because nuance source/build manager logic)